src/pages/home/report/ReportActionsView.js

```diff
@@ -278,6 +278,21 @@ class ReportActionsView extends React.Component {
         updateLastReadActionID(this.props.reportID, maxVisibleSequenceNumber);
     }
 
+    /**
+     * Checks whether given sequence number belongs to the most recent IOU Report
+     *
+     * @param {Number} actionSequenceNumber
+     * @returns {Boolean}
+     */
+    isMostRecentIOUReport(actionSequenceNumber) {
+        const mostRecentIOUReportSequenceNumber = _.chain(this.props.reportActions)
+            .sortBy('sequenceNumber')
+            .filter(action => action.actionName === 'IOU')
+            .max(action => action.sequenceNumber)
+            .value().sequenceNumber;
+        return actionSequenceNumber === mostRecentIOUReportSequenceNumber;
+    }
```

Comment:
It seems like we are performing this loop as a way of "flagging" an action as the `mostRecentIOUReportSequenceNumber`. This seems inefficient to me since we have `n` number of actions and only ONE of them will be most recent. Yet each time we render an action we will loop over all to find the number again and compare it to the action's `sequenceNumber`.

Here are some alternative ideas...

**Store this on the report instead**

- When updating a report's actions look for the most recent IOU action and set the sequenceNumber to the report object
- When updating the actions we'll update this number.

**Only calculate when the actions change**
 
Only calculate the `mostRecentSequenceNumber` when the actions change in the `updateSortedReportActions` method here:

https://github.com/Expensify/Expensify.cash/blob/3dfea0090410aed64199f2e37680429f2682ee0a/src/pages/home/report/ReportActionsView.js#L196-L203